RecoLocalTracker/SiPhase2VectorHitBuilder/interface/VectorHitMomentumHelper.h

```diff
@@ -0,0 +1,25 @@
+#ifndef VectorHitMomentumHelper_H
+#define VectorHitMomentumHelper_H
+
+#include "DataFormats/TrackerRecHit2D/interface/VectorHit.h"
+#include "MagneticField/Engine/interface/MagneticField.h"
+
+class VectorHitMomentumHelper {
+public:
+  VectorHitMomentumHelper();
+  ~VectorHitMomentumHelper() {
+  }
+
+  float transverseMomentum(VectorHit& vh,const MagneticField* magField){
+	  GlobalPoint center(0.0, 0.0, 0.0);
+	  float magnT = magField->inTesla(center).mag();
```

Comment:
1) of course is not precise
2) please use cached "nominalValue"

Reply:
I am a bit confused, the nomimalValue returns an int? https://github.com/cms-sw/cmssw/blob/1711d0fe46c07e72b3a77242f61df6fe98860fad/MagneticField/Engine/interface/MagneticField.h#L49-L53

Reply:
I get this one now. In practice this would change the magnetic field used from 3.8112 T to 3.8 T